.github/workflows/test.yaml

```diff
@@ -6,29 +6,44 @@ on:
       - main
 
 jobs:
-  test:
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-      fail-fast: false
-
-    runs-on: ${{ matrix.os }}
+  static:
+    runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
 
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: taplo-cli@0.9.3
+
       - uses: Swatinem/rust-cache@v2
 
+      - name: Toml format
+        run: make static-toml
+
       - name: Format
         run: make fmt
 
       - name: Lint
         run: make clippy
```

Comment:
Future work: There are some nice GitHub actions that do these checks but add comments and suggestions to the code instead of just failing. It might be good to look into them like https://github.com/marketplace/actions/clippy-check and https://github.com/marketplace/actions/check-formatting-of-rust-code-with-rustfmt

Reply:
My opinion is that with a project this size, I should always `make` before I push and if i forget then the time spent opening the build failure and looking at what I did wrong is a just punishment for wasting CI time :D 

The second action does look nice, as the amount of code here grows it could be useful to reviewers to see that instead of a build log